[TOPIC-BLE-LLCP] Bluetooth: controller: allow multiple version requests

kruithofa opened the pull request:
At the linked layer level we are only allowed to send one VERSION_IND
per connection, but the host may request version information
multiple times.
We need to ensure that we fill out data correctly in case of
multiple requests, even when we only send one VERSION_IND
over the air. We need to set the response_opcode correctly
since that is used for selecting how to decode/fill in correct data
for the host

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Commit:
[TOPIC-BLE-LLCP] Bluetooth: controller: allow multiple version requests

At the linked layer level we are only allowed to send one VERSION_IND
per connection, but the host may request version information
multiple times.
We need to ensure that we fill out data correctly in case of
multiple requests

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

kruithofa commented:
Add unit test for sending 2 VERSION_IND

thoh-ot reviewed (changes requested):
The unit test that you mention in the comments is missing.

kruithofa commented:
> The unit test that you mention in the comments is missing.

Indeed; I wonder where it went. In any case I'll put it back in

Commit:
[TOPIC-BLE-LLCP] Bluetooth: controller: allow multiple version requests

Added unit test for sending 2 VERSION_IND

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

kruithofa commented:
> The unit test that you mention in the comments is missing.

I found out; I added the test on the branch for implementing HCI interface. The test is now added here.
